### PR TITLE
refactor: Replace react-bootstrap MenuItems with Antd Menu

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
Replaces MenuItems in SliceHeaderControls, DisplayQueryButton and SliceAdder components.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
image

After:
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kgabryje** [comment]
@rusackas @junlincc Can you take a look?

File: superset-frontend/src/common/components/index.tsx [hunk 1]
```diff
@@ -33,8 +33,10 @@ export {
   DatePicker,
   Dropdown,
   Empty,
+  Input,
   Modal,
   Popover,
+  Select,
```
Comment (rusackas) [inline superset-frontend/src/common/components/index.tsx:39]: Are these Input/Select elements significantly different from a visual perspective? We should consider restyling them as needed to look like the SIP-34 design system, and eventually replacing the other disparate Input/Select components with these ones for consistency. Probably not a rush, but I just want to make sure we follow up on this stuff when we add new species :D

File: superset-frontend/src/components/Menu/LanguagePicker.tsx [hunk 2]
```diff
@@ -46,17 +46,23 @@ export default function LanguagePicker({
...
+      <Menu
+        onSelect={({ key }) => {
+          window.location.href = languages[key].url;
+        }}
+      >
+        {Object.keys(languages).map(langKey =>
+          langKey === locale ? null : (
+            <Menu.Item key={langKey}>
+              {' '}
+              <div className="f16">
```
Comment (rusackas) [inline superset-frontend/src/components/Menu/LanguagePicker.tsx:58]: Any idea what this `f16` class does?

File: superset-frontend/src/dashboard/components/SliceAdder.jsx [hunk 3]
```diff
@@ -52,12 +52,14 @@ const defaultProps = {
 };
 
 const KEYS_TO_FILTERS = ['slice_name', 'viz_type', 'datasource_name'];
-const KEYS_TO_SORT = [
-  { key: 'slice_name', label: 'Name' },
-  { key: 'viz_type', label: 'Vis type' },
-  { key: 'datasource_name', label: 'Dataset' },
-  { key: 'changed_on', label: 'Recent' },
-];
+const KEYS_TO_SORT = {
```
Comment (rusackas) [inline superset-frontend/src/dashboard/components/SliceAdder.jsx:55]: Nice simplification, thanks :D

File: superset-frontend/src/explore/components/controls/DateFilterControl.jsx [hunk 4]
```diff
@@ -453,66 +451,45 @@ class DateFilterControl extends React.Component {
               >
                 <div
                   className="clearfix centered"
-                  style={{ marginTop: '12px' }}
+                  style={{ marginTop: '8px', display: 'flex' }}
```
Comment (rusackas) [inline superset-frontend/src/explore/components/controls/DateFilterControl.jsx:454]: Would love to see this in Emotion with gridUnits, but not losing sleep over it.

BTW, nice to see some LESS styles disappearing with this PR :)

File: superset-frontend/src/common/components/index.tsx (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/src/common/components/index.tsx:39]: The difference is not that significant. Still, when react-bootstrap input was next to Antd Select, it looked weird.
Before:
image
After:
image

File: superset-frontend/src/explore/components/controls/DateFilterControl.jsx (hunk 4 above)
Comment (kgabryje) [reply superset-frontend/src/explore/components/controls/DateFilterControl.jsx]: Done, here and in the rest of that file

**Comment by kgabryje** [comment]
@rusackas This PR was bigger than it was supposed to be because I forgot to do the rebase and there were some changes that were already merged - I apologise for that. On the bright side, you caught some css in `style` thanks to that 😛

File: superset-frontend/src/components/Menu/LanguagePicker.tsx (hunk 2 above)
Comment (kgabryje) [reply superset-frontend/src/components/Menu/LanguagePicker.tsx]: `.f16 .flag-code` and `.f16 .flag` provide styles for the flag icons

**Review by rusackas — APPROVED** [review]
LGTM, thanks!

LGTM

**Comment by ktmud** [comment]
Previously the menu will automatically close whichever item you click, now it doesn't close.

menu-not-close

@rusackas @kgabryje

**Pull request merged**